### PR TITLE
Add --encrypt option to build command

### LINKED ISSUE #4241: Require -e flag to build encrypted containers
As the title says. This will make it clear that the build command is created an encrypted container. If we pick up env vars for for keys/passphrases, and this flag is not specified, we should give some kind of warning. Something that is not completely clear to me is if we should have this be required for `exec/shell/run` or if we should just act on envvars and other flags for those commands

### PR DESCRIPTION
**Pull request by ikaneshiro**
Signed-off-by: Ian Kaneshiro <iankane@umich.edu>

- Also adds check at runtime as specified in https://github.com/sylabs/singularity/issues/4241#issuecomment-522734611

**This fixes or addresses the following GitHub issues:**

- Fixes #4241

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by schebro — APPROVED** [review]

File: cmd/internal/cli/actions_linux.go [hunk 1]
```diff
@@ -258,12 +258,32 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
...
 	}
+	defer img.File.Close()
+
+	if !img.HasRootFs() {
+		sylog.Fatalf("no root filesystem found in %s", engineConfig.GetImage())
+	}
+
+	// ensure we have decryption material
+	if img.Partitions[0].Type == imgutil.ENCRYPTSQUASHFS {
+		sylog.Debugf("Encrypted container filesystem detected")
```
Comment (schebro) [inline cmd/internal/cli/actions_linux.go]: Encrypted filesystem detected?

File: cmd/internal/cli/actions_linux.go [hunk 2]
```diff
@@ -258,12 +258,32 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
...
+		if encryptionKey == "" {
```
Comment (schebro) [inline cmd/internal/cli/actions_linux.go]: encryptionKey is retried on line 280, right? Won't we hit this Fatalf always?

File: cmd/internal/cli/build_linux.go [hunk 3]
```diff
@@ -186,6 +191,17 @@ func run(cmd *cobra.Command, args []string) {
...
 	} else {
+		// ensure passphrase or key was supplied with encrypt option
+		if encrypt && !cmd.Flags().Lookup("encryption-key").Changed {
+			sylog.Fatalf("Unable to encrypt container, no passphrase or key path specified.")
+		}
+
+		// ensure we do not build an encrypted container if encrypt option is not specified
+		if !encrypt && cmd.Flags().Lookup("encryption-key").Changed {
+			sylog.Warningf("Encryption key environment variable found, but -e was not specified. NOT encrypting container.")
+			encryptionKey = ""
```
Comment (schebro) [inline cmd/internal/cli/build_linux.go:202]: Where is this key used?

Comment (ikaneshiro) [reply cmd/internal/cli/build_linux.go:202]: It is in the build config below as it was before this change. This is just preventing an encrypted container build from happening if `-e` isnt there

File: cmd/internal/cli/actions_linux.go (hunk 2 above)
Comment (ikaneshiro) [reply cmd/internal/cli/actions_linux.go]: No this is just catching the case of no key material upfront. I think as is we might allow for an encrypted container with an empty passphrase?

File: cmd/internal/cli/actions_linux.go (hunk 1 above)
Comment (ikaneshiro) [reply cmd/internal/cli/actions_linux.go]: I think its better to be slightly more verbose here

File: cmd/internal/cli/actions_linux.go (hunk 2 above)
Comment (schebro) [reply cmd/internal/cli/actions_linux.go]: Didn't think about that case. It's fine for now I guess.

**Review by schebro — APPROVED** [review]

**Pull request merged**